### PR TITLE
Add support for weave password in RKE

### PR DESCRIPTION
**Pull request by codezomb**
Adds support for weave encryption in RKE - https://github.com/rancher/rke/pull/781#issuecomment-406032648

### CODE REVIEW

File: apis/management.cattle.io/v3/rke_types.go [hunk 1]
```diff
@@ -371,6 +373,10 @@ type CanalNetworkProvider struct {
 	FlannelNetworkProvider `yaml:",inline" json:",inline"`
 }
 
+type WeaveNetworkProvider struct {
+	Password string `yaml:",omitempty" json:",omitempty"`
```
[the remark this answers is not in the document]
Comment (codezomb) [reply apis/management.cattle.io/v3/rke_types.go]: It's been a while so I don't exactly remember. I'm pretty new to Go so there was a bit of trial and error and digging around other examples. If there's a better way, I'm all for it.

[the remark this answers is not in the document]
Comment (alena1108) [reply apis/management.cattle.io/v3/rke_types.go]: @codezomb lets just specify `password` in annotations to be on the safe side

[the remark this answers is not in the document]
Comment (codezomb) [reply apis/management.cattle.io/v3/rke_types.go]: No problem! Just to be clear, you would prefer this?

```Go
type WeaveNetworkProvider struct {
	Password string `yaml:"password,omitempty" json:"password,omitempty"`
}
```

[the remark this answers is not in the document]
Comment (alena1108) [reply apis/management.cattle.io/v3/rke_types.go]: @codezomb do you think we can update the pr any time soon? I'd like to merge it

**Comment by codezomb** [comment]
Sorry, I’ll get it today.

--
Mike Kelley
Sent with Airmail

**Comment by codezomb** [comment]
@alena1108 I've pushed the update, sorry for the delay!

**Comment by alena1108** [comment]
@codezomb looks like drone is not happy, could you check? Also if you could squash 2 functional commits into 1 - and have one extra commit with the generated code - that would be great.

**Pull request merged**